File: ios/AirshipFrameworkProxyTests/ScopedSubscriptionListOperationTest.swift
import XCTest

@testable import AirshipFrameworkProxy
import AirshipKit

class ScopedSubscriptionListOperationTest: XCTestCase {

    func testParse() throws {
        let json = """
            {
                "listId": "some list",
                "action": "subscribe",
                "scope": "app"
            }
        """


        let parsed = try JSONDecoder().decode(ScopedSubscriptionListOperation.self, from: json.data(using: .utf8)!)

        let expected = ScopedSubscriptionListOperation(
            action: .subscribe,
            listID: "some list",
            scope: .app
        )

        XCTAssertEqual(expected, parsed)
    }

    func testParseInvalidScope() throws {
        let json = """
           {
               "listId": "some list",
               "action": "subscribe",
               "scope": "invalid"
           }
        """

        do {
            let _ = try JSONDecoder().decode(ScopedSubscriptionListOperation.self, from: json.data(using: .utf8)!)
            XCTFail("should throw")
        } catch {}
    }

}

File: android/airship-framework-proxy/src/test/java/com/urbanairship/android/framework/proxy/ScopedSubscriptionListOperationTest.kt

package com.urbanairship.android.framework.proxy

import com.urbanairship.contacts.Scope
import com.urbanairship.json.JsonValue
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import java.util.*

@RunWith(RobolectricTestRunner::class)
public class ScopedSubscriptionListOperationTest {

    @Test
    public fun testParse() {
        val json = """
            {
                "listId": "some list",
                "action": "subscribe",
                "scope": "app"
            }
        """

        val parsed = ScopedSubscriptionListOperation(json = JsonValue.parseString(json).requireMap())

        val expected = ScopedSubscriptionListOperation(
            action = SubscriptionListOperationAction.SUBSCRIBE,
            scope = Scope.APP,
            listId = "some list"
        )

        assertEquals(expected, parsed)
    }

    @Test(expected = IllegalArgumentException::class)
    public fun testParseInvalidScope() {
        val json = """
           {
               "listId": "some list",
               "action": "subscribe",
               "scope": "invalid"
           }
        """

        ScopedSubscriptionListOperation(json = JsonValue.parseString(json).requireMap())
    }
}
